core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2175,6 +2182,20 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     })
   }
 
+  // A new workspace request may specify the region where the bucket should be created. In the case of cloning a
+  // workspace, if no bucket location is provided, then the cloned workspace's bucket will be created in the same region
+  // as the source workspace's bucket. Rawls does not store bucket regions, so in order to get this information we need
+  // to query Google and this query costs money, so we need to make sure that the target Google Project is the one that
+  // gets charged. If neither a bucket location nor a source bucket name are provided, Future[None] is returned which
+  // will result in the default bucket location being used
+  private def determineWorkspaceBucketLocation(maybeBucketLocation: Option[String], maybeSourceBucketName: Option[String], googleProjectId: GoogleProjectId): Future[Option[String]] = {
+    (maybeBucketLocation, maybeSourceBucketName) match {
+      case (bucketLocation@Some(_), _) => Future(bucketLocation)
```

Comment:
would 
```
case (Some(bucketLocation), _) => Future(bucketLocation)
```
not work here?

Reply:
Nope, the `@` symbol sets the variable name before it to be equal to whatever matches the pattern after it. So in the case of `bucketLocation@Some(_)`, `bucketLocation` is of type `Option[String]` which is the desired type. 

with
```
case (Some(bucketLocation), _) => Future(bucketLocation)
```

`bucketLocation` would be type `String` and we'd be returning `Future[String]` rather than `Future[Option[String]]`

Reply:
Ok that makes sense. So if I'm understanding it correctly, would this work then?

```
case (Some(bucketLocation), _) => Future(Option(bucketLocation))
```

Reply:
Yes, that would work as well